celery/worker/worker.py

```diff
@@ -158,7 +158,11 @@ def on_close(self):
 
     def on_stopped(self):
         self.timer.stop()
-        self.consumer.shutdown()
+        if isinstance(self.consumer, list):
```

Comment:
Instead of adjusting the `w.consumer` to be a list, use the new flag to choose which consumer to act upon